### PR TITLE
add system locale check

### PR DESCRIPTION
**Pull request by mattma**
add system locale check. render the respond locale based on user's default locale.

### CODE REVIEW

**Comment by mattma** [comment]
since the server starts once, it checks once. it won't run on every check when there is an user connection. so i guess it won't work, right?

But if it runs every route match, it will waste too much bandwidth for no reason.

File: app/router.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+
 module.exports = app => {
   app.get('/zh-cn*', 'cn');
   app.get('/en*', 'en');
-  app.redirect('/', '/en');
+
+  // redirect based on User's system locale
+  app.redirect('/', defaultLocale);
+  // Catch all clause. Trigger when there is no match
+  app.get('(.*)', defaultLocale);
```
Comment (atian25) [inline app/router.js:15]: `router.js` will only exec once at start time.

File: app/router.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
 'use strict';
 
+const osLocale = require('os-locale');
+const locale = osLocale.sync();
+// China zh_CN
+const defaultLocale = locale === 'en_US' ? '/en': '/zh-cn';
```
Comment (atian25) [inline app/router.js:6]: so you want to set default locale with machine locale settings?

Comment (mattma) [reply app/router.js:6]: I guess that is not what I was trying to do. It should handle at the Client side.

Comment (atian25) [reply app/router.js:6]: yes, so I don't think you should use this. you should detect by header or query

Comment (mattma) [reply app/router.js:6]: Totally agree. I will cancel this PR.

**Pull request closed without merging**